services/tenant-ui/frontend/src/store/tenantStore.ts

```diff
@@ -62,105 +66,183 @@ export const useTenantStore = defineStore('tenant', () => {
+    loadingIssuance.value = true;
 
-    await tenantApi
-      .getHttp(API_PATH.TENANT_CONFIGURATION)
-      .then((res) => {
-        console.log(res);
-        tenantConfig.value = res.data.item;
-        console.log(tenant.value);
+    await acapyApi
+      .getHttp(API_PATH.TENANT_ENDORSER_INFO)
+      .then((res: any) => {
+        endorserInfo.value = res.data;
       })
       .catch((err) => {
         error.value = err;
-        tenantConfig.value = null;
+        endorserInfo.value = null;
       })
       .finally(() => {
-        loading.value = false;
+        loadingIssuance.value = false;
       });
-    console.log('< tenantStore.getConfiguration');
+    console.log('< tenantStore.getEndorserInfo');
 
     if (error.value != null) {
       // throw error so $onAction.onError listeners can add their own handler
       throw error.value;
     }
     // return data so $onAction.after listeners can add their own handler
-    return tenantConfig.value;
+    return endorserInfo.value;
   }
 
-  async function updateConfiguration(payload: any = {}) {
-    console.log('> tenantStore.updateConfiguration');
+  async function connectToEndorser() {
+    console.log('> contactsStore.createInvitation');
     error.value = null;
-    loading.value = true;
-    console.log(payload);
-    await tenantApi
-      .putHttp(API_PATH.TENANT_CONFIGURATION, payload)
+    loadingIssuance.value = true;
+
+    await acapyApi
+      .postHttp(API_PATH.TENANT_ENDORSER_CONNECTION, {})
       .then((res) => {
         console.log(res);
-        tenantConfig.value = res.data.item;
-        console.log(tenant.value);
+        endorserConnection.value = res.data;
       })
       .catch((err) => {
         error.value = err;
-        console.log(error.value);
       })
       .finally(() => {
-        loading.value = false;
+        loadingIssuance.value = false;
       });
-    console.log('< tenantStore.updateConfiguration');
+    console.log('< contactsStore.connectToEndorser');
+
+    if (error.value != null) {
+      // throw error so $onAction.onError listeners can add their own handler
+      throw error.value;
+    }
+  }
+
+  async function getPublicDid() {
+    console.log('> tenantStore.getPublicDid');
+    error.value = null;
+    loadingIssuance.value = true;
+
+    await acapyApi
+      .getHttp(API_PATH.WALLET_DID_PUBLIC)
+      .then((res: any) => {
+        publicDid.value = res.data;
+      })
+      .catch((err) => {
+        error.value = err;
+        publicDid.value = null;
+      })
+      .finally(() => {
+        loadingIssuance.value = false;
+      });
+    console.log('< tenantStore.getPublicDid');
 
     if (error.value != null) {
       // throw error so $onAction.onError listeners can add their own handler
       throw error.value;
     }
     // return data so $onAction.after listeners can add their own handler
-    return tenantConfig.value;
+    return publicDid.value;
+  }
+
+  async function registerPublicDid() {
```

Comment:
This is great. Only thing I can think of (not necessary right now) is to break this up in the service/store, and have the "client" vue file call this sequence with pauses and allow it to update the screen about which step it is on. Not sure if we have any facility in the spinner to provide "live" feedback.

The flow and screens look good to me.

Reply:
Yeah breaking it up into discrete user actions is a UX (and error handling) decision we can iterate on.
Great idea about the screen text @usingtechnology , I've updated that like below for the multi-stage registration one

![image](https://user-images.githubusercontent.com/17445138/216199629-e2d96bab-b28f-450c-b064-d4e8ad8ca867.png)

commit:  https://github.com/bcgov/traction/pull/425/commits/39469fc4afa77fab4cf4c0bc6842d408e350b482